ddtrace/internal/telemetry/writer.py

```diff
@@ -123,11 +159,151 @@ def _flush_events_queue(self):
             self._events_queue = []
         return events
 
+    def enable(self):
+        # type: () -> None
+        """
+        Enable the instrumentation telemetry collection service. If the service has already been
+        activated before, this method does nothing. Use ``disable`` to turn off the telemetry collection service.
+        """
+        if self.status == ServiceStatus.RUNNING:
+            return
+
+        self._enabled = True
+        self.start()
+
+        atexit.register(self.stop)
+
+    def _stop_service(self, *args, **kwargs):
+        # type: (...) -> None
+        super(TelemetryBase, self)._stop_service(*args, **kwargs)
+        self.join()
+
+    def _fork_writer(self):
+        # type: () -> None
+        pass
+
+
+class TelemetryMetricsWriter(TelemetryBase):
+    """
+    Submits Telemetry Metrics events to the datadog agent.
+    """
+
+    def __init__(self):
+        # type: () -> None
+        super(TelemetryMetricsWriter, self).__init__(interval=_get_telemetry_metrics_interval_or_default())
+        self._namespace = MetricNamespace()
+
+    def _flush_namespace_metrics(self):
+        # type () -> List[Metric]
+        """Returns a list of all generated metrics and clears the namespace's list"""
+        log.debug("collect Telemetry Metrics")
+        with self._lock:
+            namespace_metrics = self._namespace.get()
+            self._namespace._flush()
+        return namespace_metrics
+
+    def _fork_writer(self):
+        # type: () -> None
+        self._namespace._flush()
+
+    def _app_generate_metrics_event(self, namespace_metrics):
+        # type: (NamespaceMetricType) -> None
+        for payload_type, namespaces in namespace_metrics.items():
+            for namespace, metrics in namespaces.items():
+                if metrics:
+                    payload = {
+                        "namespace": namespace,
+                        "lib_language": "python",
+                        "lib_version": _pep440_to_semver(),
+                        "series": [m.to_dict() for m in metrics.values()],
+                    }
```

Comment:
Random thought (feel free to ignore); We can add more detail to this log message. Also what’s the expect size of the payloads. Could this significantly increase the size of a users debug logs? Could we print a few fields instead of the full payload?